### ISSUE TITLE
Dont read response content on network error.

### ISSUE THREAD

**Issue by Tetragramat** [h4kuna/fio#57]
When Fio movements ends with network error, the \h4kuna\Fio\Read\Json::create reads content anyway and ends throwing RuntimeException "Unable to read stream contents".

**Comment by h4kuna** [comment]
What behavior do you expect? I haven't RuntimeException class in my library and nette throw JsonException. What do you have library for communication?

**Comment by Tetragramat** [comment]
It's psr 7 behavior this package is using. 
https://github.com/php-fig/http-message/blob/master/src/StreamInterface.php#L140

**Comment by h4kuna** [comment]
What behavior do you expect?

**Comment by Tetragramat** [comment]
I'm using `Symfony\Component\HttpClient\Psr18Client`

The whole error is
`[RuntimeException] Unable to read stream contents: Idle timeout reached for "https://fioapi.fio.cz/v1/rest/periods/REDACTED/2024-06-06/2024-06-06/transactions.json".`

I would expect \h4kuna\Fio\Exceptions\ServiceUnavailable that should be thrown.

**Comment by Tetragramat** [comment]
What exactly happens is that Fio starts streaming response, but then timeout occurs. So status code is 200, but when you call getContents it returns timeout.

**Comment by h4kuna** [comment]
Ok I understand. I saw to sentry on Fio, where we are using guzzle and i find. 
```
h4kuna\Fio\Exceptions\ServiceUnavailable
 
/app/Finance/Fio/Services/FioService.php in App\Finance\Fio\Services\FioService::sinceDate

Level: Error
Idle timeout reached for "https://www.fio.cz/ib_api/rest/periods/REDACTED/2024-03-13/2024-03-13/transactions.json".
```

**Comment by h4kuna** [comment]
Can you try `dev-master` a few weeks?

**Comment by Tetragramat** [comment]
It happen quite often so I can try it replicate locally, but probably earliest on monday.

**Comment by Tetragramat** [comment]
I can't use dev-master on production. Can't try locally too because the issue with fio was only 5-7th jun. Nothing since. I'll keep watch if it happens again in next days.

**Comment by Tetragramat** [comment]
Anyway looking at the fix https://github.com/h4kuna/fio/commit/0406b9bc152fc6b107856b1dbf6bb466a56d3dff, you should add test for this issue.

**Comment by Tetragramat** [comment]
It's happening again and so I tested master branch locally.

I got 200 with part of the response content and then timeout. 
This library returned `[h4kuna\Fio\Exceptions\ServiceUnavailable] Unable to read stream contents: Idle timeout reached for "REDACTED"`.

**Comment by h4kuna** [comment]
Is it mean the right behaviour?

**Comment by Tetragramat** [comment]
yes, it works as expected.

**Comment by h4kuna** [comment]
Released 3.0.10
